SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultWorkerJsonGenerator.cpp

```diff
@@ -46,12 +46,19 @@ bool GenerateAllDefaultWorkerJsons(bool& bOutRedeployRequired)
 
 	if (const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>())
 	{
-		const FName WorkerTypes[] = { SpatialConstants::DefaultServerWorkerType, SpatialConstants::RoutingWorkerType,
-									  SpatialConstants::StrategyWorkerType };
-		for (auto Worker : WorkerTypes)
+		// Create an array of worker types with a bool signifying whether the associated worker json should exist or not.
+		// Then correct the file system state so it matches our expectations.
+		TArray<TPair<FName, bool>> WorkerTypes;
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::DefaultServerWorkerType, true));
+		const bool bRoutingWorkerEnabled = SpatialGDKSettings->CrossServerRPCImplementation == ECrossServerRPCImplementation::RoutingWorker;
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::RoutingWorkerType, bRoutingWorkerEnabled));
+		WorkerTypes.Add(TPair<FName, bool>(SpatialConstants::StrategyWorkerType, SpatialGDKSettings->bRunStrategyWorker));
+
+		for (const auto& Pair : WorkerTypes)
 		{
-			FString JsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.json"), *Worker.ToString()));
-			if (!FPaths::FileExists(JsonPath))
+			FString JsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.json"), *Pair.Key.ToString()));
```

Comment:
could pull this out as a static member?

Reply:
Hm, not sure what you mean

Reply:
the `spatialos.%s.worker.json` string?

Reply:
So, I have tried this, and it actually seems kind of hard to do.
Printf is giving me some issues if I move this out.